### PR TITLE
MGDOBR-40: use org.assertj.core.api.Assertions in all tests

### PR DESCRIPTION
**Pull request by kostola**
* [Jira task](https://issues.redhat.com/browse/MGDOBR-40)

### CODE REVIEW

**Review by r00ta — APPROVED** [review]
Thank you!

**Review by robpblake — APPROVED** [review]
@kostola What is the SonarLint configuration you are using here? If we're going to be linting, then we should have it in the project configuration and CI pipeline to make sure that everyone is doing it consistently.

**Comment by kostola** [comment]
> @kostola What is the SonarLint configuration you are using here? If we're going to be linting, then we should have it in the project configuration and CI pipeline to make sure that everyone is doing it consistently.

@robpblake I guess it's the default one used by the IntelliJ SonarLint plugin, since I didn't configure it out of the default settings for this project. I made those changes without thinking too much since I was already reviewing tests and I got the highlights in my IDE. Do you want me to revert them while we decide how to configure SonarLint?

**Comment by robpblake** [comment]
> > @kostola What is the SonarLint configuration you are using here? If we're going to be linting, then we should have it in the project configuration and CI pipeline to make sure that everyone is doing it consistently.
> 
> @robpblake I guess it's the default one used by the IntelliJ SonarLint plugin, since I didn't configure it out of the default settings for this project. I made those changes without thinking too much since I was already reviewing tests and I got the highlights in my IDE. Do you want me to revert them while we decide how to configure SonarLint?

Yeah I think let's keep the changes here focused on migrating to `org.assertj.core.api.Assertions` only. 

Can you create an issue in our backlog to agree on and apply Linting. I'm happy for us to agree on SonarLint if it's easy to use and most importantly, we can enforce it in the CI pipeline. Cheers 👍

File: actions/src/test/com/redhat/service/bridge/actions/kafkatopic/KafkaTopicActionValidatorTest.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
 import com.redhat.service.bridge.infra.models.dto.BridgeDTO;
 import com.redhat.service.bridge.infra.models.dto.ProcessorDTO;
 import io.quarkus.test.junit.QuarkusTest;
-import org.junit.jupiter.api.Assertions;
+import static org.assertj.core.api.Assertions.assertThat;
```
Comment (danielezonca) [inline actions/src/test/com/redhat/service/bridge/actions/kafkatopic/KafkaTopicActionValidatorTest.java:13]: Please check formatting

@r00ta 
As far as I remember reformat should also work for import static. Am I wrong?

**Comment by kostola** [comment]
@robpblake I reverted the changes regarding SonarLint :+1:

File: actions/src/test/com/redhat/service/bridge/actions/kafkatopic/KafkaTopicInvokerTest.java [hunk 2]
```diff
@@ -7,7 +7,7 @@
 import org.eclipse.microprofile.reactive.messaging.Emitter;
 import org.eclipse.microprofile.reactive.messaging.Message;
 import org.eclipse.microprofile.reactive.messaging.Metadata;
-import org.junit.jupiter.api.Assertions;
+import static org.assertj.core.api.Assertions.assertThat;
```
Comment (danielezonca) [inline actions/src/test/com/redhat/service/bridge/actions/kafkatopic/KafkaTopicInvokerTest.java:10]: Same here

File: manager/src/test/java/com/redhat/service/bridge/manager/api/user/ProcessorAPITest.java [hunk 3]
```diff
@@ -29,13 +28,7 @@
 import io.quarkus.test.junit.QuarkusTest;
 import io.restassured.response.Response;
 
-import static java.util.Arrays.asList;
-import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.equalTo;
```
Comment (danielezonca) [inline manager/src/test/java/com/redhat/service/bridge/manager/api/user/ProcessorAPITest.java:34]: Can you remove `org.hamcrest` from build-parent? Now it should have no remaining usage right?

File: actions/src/test/com/redhat/service/bridge/actions/kafkatopic/KafkaTopicActionValidatorTest.java (hunk 1 above)
Comment (kostola) [reply actions/src/test/com/redhat/service/bridge/actions/kafkatopic/KafkaTopicActionValidatorTest.java:17]: Fixed :+1:

File: actions/src/test/com/redhat/service/bridge/actions/kafkatopic/KafkaTopicInvokerTest.java (hunk 2 above)
Comment (kostola) [reply actions/src/test/com/redhat/service/bridge/actions/kafkatopic/KafkaTopicInvokerTest.java]: Fixed :+1:

File: manager/src/test/java/com/redhat/service/bridge/manager/api/user/ProcessorAPITest.java (hunk 3 above)
Comment (kostola) [reply manager/src/test/java/com/redhat/service/bridge/manager/api/user/ProcessorAPITest.java:34]: Hamcrest is currently not imported in any `pom.xml` file.
However, it's a dependency of `awaitility`, which uses Hamcrest `Matchers` in some APIs like, for example, to assert the content of the response body. These are currently used only in [End2EndTestIT.java](https://github.com/kostola/sandbox/blob/MGDOBR-40/runner/src/test/java/com/redhat/service/bridge/runner/it/End2EndTestIT.java).

**Comment by danielezonca** [comment]
@kostola please check test failure

**Comment by kostola** [comment]
@danielezonca build is ok now :+1: (I guess it was some internal error because I retriggered it without changing anything)

**Pull request merged**